geomesa-arrow/geomesa-arrow-jts/src/main/java/org/locationtech/geomesa/arrow/vector/WKBGeometryVector.java

```diff
@@ -0,0 +1,109 @@
+/***********************************************************************
+ * Copyright (c) 2013-2020 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.arrow.vector;
+
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.vector.Float8Vector;
+import org.apache.arrow.vector.VarBinaryVector;
+import org.apache.arrow.vector.complex.AbstractContainerVector;
+import org.apache.arrow.vector.complex.FixedSizeListVector;
+import org.apache.arrow.vector.types.pojo.ArrowType;
+import org.apache.arrow.vector.types.pojo.Field;
+import org.apache.arrow.vector.types.pojo.FieldType;
+import org.locationtech.geomesa.arrow.vector.impl.AbstractPointVector;
+import org.locationtech.jts.geom.Geometry;
+import org.locationtech.jts.io.ParseException;
+import org.locationtech.jts.io.WKBReader;
+import org.locationtech.jts.io.WKBWriter;
+
+import javax.annotation.Nullable;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Catch-all for storing instances of Geometry as WKB
+ */
+public class WKBGeometryVector implements GeometryVector<Geometry, VarBinaryVector> {
+  private VarBinaryVector vector;
+  private WKBWriter writer = new WKBWriter();
+  private WKBReader reader = new WKBReader();
+
+  public static FieldType createFieldType(Map<String, String> metadata) {
+    return new FieldType(true, ArrowType.Binary.INSTANCE, null, metadata);
+  }
+
+  public WKBGeometryVector(String name, BufferAllocator allocator, @Nullable Map<String, String> metadata) {
+    // JNH: How do I use the metadata?
+    this.vector = new VarBinaryVector(name, createFieldType(metadata), allocator);
+  }
+
+  public WKBGeometryVector(String name, AbstractContainerVector container, @Nullable Map<String, String> metadata) {
+    this.vector = container.addOrGet(name, createFieldType(metadata), VarBinaryVector.class);
+  }
+
+  public WKBGeometryVector(VarBinaryVector vector) {
+    this.vector = vector;
+  }
+
+  @Override
+  public void set(int i, Geometry geom) {
+    if (geom == null) {
+      vector.setNull(i);
+    } else {
+
+      vector.setSafe(i, writer.write(geom));
+    }
+  }
+
+  @Override
+  public Geometry get(int i) {
+    if (vector.isNull(i)) {
+      return null;
+    } else {
+      Geometry geometry = null;
+      try {
+         geometry = reader.read(vector.get(i));
+      } catch (ParseException exception) {
+        // JNH: Swallow exception?!
```

Comment:
nooooooooo :scream:

Reply:
It seems like the only option here is to log it and move on.  

The API doesn't have this method throwing an exception.

Reply:
ah, yeah java typed exceptions. you could throw a runtime exception though, i think that would probably be the best solution, then we can handle it higher up the chain

Reply:
Ok, so re-throwing as a Runtime exception makes sense?

Reply:
Done